tests/std/tests/P0896R4_ranges_iterator_machinery/test.cpp

```diff
@@ -1877,10 +1877,11 @@ namespace iter_ops {
 
     enum class sized { no, yes };
     enum class assign { no, yes };
+    enum class nothrow { no, yes };

```

Comment:
Why must these be enum classes instead of simple bools?

It is fun to read `nothrow NoThrow = nothrow::no` (no, NoThrow!) but `noexcept(IsNoThrow)` might be clearer...

Pre-existing

Reply:
Because it is much more difficult to reason about what `meow<true>` does compared to `meow<nothrow::no>`

Also it is much harder to accidentally pass the wrong thing

Reply:
Especially in this case where it is
```cpp
trace_iterator<Cat, true, false, true>
```

Are you sure those are in the right order?

Reply:
`nothrow` is a terrible name. How many times have I told someone not to put a negative in a name?

Reply:
But you told ***them*** not to do it. 

Nobody said **you** could not commit horrible code crimes

Reply:
May I suggest something giving the best of both worlds (in the future, this is pre-existing, also using enum helps a ton):
```c++
enum : bool { Throws, NoThrow };
template<bool IsNoThrow = false>
struct meow { ... };

meow<NoThrow> ....
```

Reply:
I suppose you could name the enum if you really want to, but that isn't relevant for clarity, just for preventing type confusion, but type confusion is arguably not a big deal here since really the type is bool for all of them.